### PR TITLE
feat: Update login flow and UI to handle first and last name display

### PR DESCRIPTION
**Pull request by MarioDanilo0111**
- Updated :
  - Added logic to store  and  in the login state.
  - Enhanced state logging for easier debugging of login-related data.

- Modified :
  - Captured  and  from the API response and set them in the login context.
  - Improved error handling for login failures.

- Updated :
  - Passed  and  to  to display user information in the header.
  - Handled cases where user information might be missing or undefined.

- Refactored :
  - Displayed  and  in the user avatar area.
  - Ensured the dropdown and logout functionality work as expected.
  - Added fallback for missing user information.

- Fixed :
  - Defined  helper function to format and display user’s full name in the page title.
  - Removed references to undefined functions and fixed errors in rendering user name.

- Updated :
  - Corrected login response to ensure  and  are populated for each user role (user, host, caseworker).
  - Enhanced mock API to return the correct  and  for specific users.

- Miscellaneous:
  - Added PropTypes validation in components where applicable.
  - Removed unnecessary  requirement in  component to resolve PropTypes warning.

### CODE REVIEW

File: src/components/Header/PageTitle.jsx [hunk 1]
```diff
@@ -3,28 +3,34 @@ import useLogin from "./../../hooks/useLogin";
...
+    return `${firstNameFormatted} ${lastNameFormatted}`;
+  };
+  return (
+    <div className="flex items-center justify-between bg-white">
+      <div className="my-6 text-3xl sm:mb-0 lg:flex justify-center font-bold">
+        {login.first_name && login.last_name
+          ? getName(login.first_name, login.last_name)
+          : "Välkommen"}
+
+        {/* {login.username != null ? getName(login?.username) : "Välkommen"} */}
```
Comment (jani-at-noq) [inline src/components/Header/PageTitle.jsx:26]: Ta borta kommenterad kod

File: src/context/LoginProvider.jsx [hunk 2]
```diff
@@ -1,20 +1,50 @@
-import { createContext, useState } from "react";
-import PropTypes from 'prop-types';
+import axios from "axios";
+import { createContext, useState, useEffect } from "react";
```
Comment (jani-at-noq) [inline src/context/LoginProvider.jsx:2]: ta borta useEffect

File: src/context/LoginProvider.jsx [hunk 3]
```diff
@@ -1,20 +1,50 @@
...
-    return (
-        <LoginContext.Provider value={{ login, setLogin }}>
-            {children}
-        </LoginContext.Provider>
-    )
-}
+  // Add handleLogin to update the state
+  const handleLogin = async (loginData) => {
+    try {
+      const response = await axios.post("/api/api/login/", loginData); // mock API call
```
Comment (jani-at-noq) [inline src/context/LoginProvider.jsx:19]: ta borta kommentar om mock API call. Detta är API call, oavsett om vi kör till mock eller backend. Ingen skillnad

File: src/context/LoginProvider.jsx [hunk 4]
```diff
@@ -1,20 +1,50 @@
...
+        last_name: data.last_name,
+        usergroups: data.groups,
+        host: data.host,
+      });
+    } catch (error) {
+      console.error("Login failed:", error);
+    }
+  };
+
+  /* useEffect(() => {
```
Comment (jani-at-noq) [inline src/context/LoginProvider.jsx:35]: ta bort

File: src/pages/HostPage.jsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React, { useEffect } from "react";
-import axios from "./../api/AxiosNoqApi"
+import axios from "./../api/AxiosNoqApi";
```
Comment (jani-at-noq) [inline src/pages/HostPage.jsx:2]: bra upptäckt

**Review by jani-at-noq — CHANGES_REQUESTED** [review]
Ser bra ut. Några lint issues behöver fixas och kanske logik fel i ett if sats.

File: src/api/mockApi/mockApi.js [hunk 6]
```diff
@@ -1,169 +1,190 @@
...
+    login.first_name = "Danilo";
+    login.last_name = "Fernandez";
+    return [200, JSON.stringify(login)];
+  }
+  // Host: user.host@test.nu
+  else if (
+    data.email == "user.host@test.nu" &&
+    data.password == "P4ssw0rd_for_Te5t+User"
+  ) {
+    login.first_name = "HostTest";
```
Comment (jani-at-noq) [inline src/api/mockApi/mockApi.js:55]: first_name: HostTest -> Hosttest eller bara Host
last_name: UserTest -> Usertest eller bara User

File: src/api/mockApi/mockApi.js [hunk 7]
```diff
@@ -1,169 +1,190 @@
...
+    login.groups = ["host"];
+    login.host = hostInfo;
+    return [200, JSON.stringify(login)];
+  }
+  // Caseworker: user.caseworker@test.nu
+  else if (
+    data.email == "user.caseworker@test.nu" &&
+    data.password == "P4ssw0rd_for_Te5t+User"
+  ) {
+    login.first_name = "CaseTest";
```
Comment (jani-at-noq) [inline src/api/mockApi/mockApi.js:66]: first_name: CaseTest -> Casetest
last_name: WorkerTest -> Worker eller Workertest

File: src/components/Common/Card.jsx [hunk 8]
```diff
@@ -4,20 +4,33 @@ import PropTypes from "prop-types";
 const Card = ({ title, content, icon }) => {
   Card.propTypes = {
     title: PropTypes.string.isRequired,
-    content: PropTypes.string.isRequired,
-    textColor: PropTypes.string.isRequired,
-    unit: PropTypes.string.isRequired,
+    content: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
```
[the remark this answers is not in the document]
Comment (MarioDanilo0111) [reply src/components/Common/Card.jsx:7]: Får "Warning" meddelande på `Card` komponent, `Failed prop type: Invalid prop `content` of type `number` supplied to `Card`, expected `string`.`

**Review by natischmidt — CHANGES_REQUESTED** [review]
behövs några fix, reqeusta min review igen efter! :)

**Review by natischmidt — APPROVED** [review]

**Pull request merged**